### PR TITLE
Avoid calling time(nullptr).

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR
@vekterli FYI

### CODE REVIEW

**Comment by baldersheim** [comment]
@geirst PR

**Comment by baldersheim** [comment]
Build failure is unrelated.

File: fastos/src/vespa/fastos/timestamp.cpp [hunk 1]
```diff
@@ -39,4 +42,8 @@ int64_t ClockSystem::now()
     return ns;
 }
 
+time_t time() {
+    return duration_cast<seconds>(system_clock::now().time_since_epoch()).count();
```
Comment (havardpe) [inline fastos/src/vespa/fastos/timestamp.cpp]: consider using the 'to_time_t' function

File: metrics/src/vespa/metrics/metricmanager.h [hunk 2]
```diff
@@ -70,6 +70,7 @@ class MetricManager : private document::Runnable
 
     struct Timer {
         virtual ~Timer() {}
+        static time_t secondsSinceEpoch();
```
Comment (havardpe) [inline metrics/src/vespa/metrics/metricmanager.h]: consider just using fastos::time() instead

File: storage/src/vespa/storage/config/distributorconfiguration.cpp [hunk 3]
```diff
@@ -123,7 +124,7 @@ DistributorConfiguration::configure(const vespa::config::content::core::StorDist
         // Always changes.
         _lastGarbageCollectionChange = 1;
     } else if (_garbageCollectionSelection != config.garbagecollection.selectiontoremove) {
-        _lastGarbageCollectionChange = time(NULL);
+        _lastGarbageCollectionChange = duration_cast<seconds>(system_clock::now().time_since_epoch()).count();
```
Comment (havardpe) [inline storage/src/vespa/storage/config/distributorconfiguration.cpp]: use fastos::time()

**Review by havardpe — COMMENTED** [review]
some minor comments

File: fastos/src/vespa/fastos/timestamp.cpp (hunk 1 above)
Comment (baldersheim) [reply fastos/src/vespa/fastos/timestamp.cpp]: Fixed

File: storage/src/vespa/storage/config/distributorconfiguration.cpp (hunk 3 above)
Comment (baldersheim) [reply storage/src/vespa/storage/config/distributorconfiguration.cpp]: Fixed

File: metrics/src/vespa/metrics/metricmanager.h (hunk 2 above)
Comment (baldersheim) [reply metrics/src/vespa/metrics/metricmanager.h]: Fixed

**Pull request merged**